code/game/objects/items/weapons/ecigs.dm

```diff
@@ -0,0 +1,296 @@
+/obj/item/clothing/mask/smokable/ecig
+	name = "electronic cigarette"
+	desc = "Device with modern approach to smoking."
+	icon = 'icons/obj/ecig.dmi'
+	var/active = FALSE
+	var/obj/item/cell/cig_cell
+	var/cell_type = /obj/item/cell/crap/cig
+	var/cartridge_type = /obj/item/reagent_containers/ecig_cartridge/med_nicotine
```

Comment:
put this beneath the ec_cartridge var so it's in the same order as the cell vars